Tools/clinic/clinic.py

```diff
@@ -4840,6 +4830,24 @@ def state_dsl_start(self, line: str) -> None:
 
         self.next(self.state_modulename_name, line)
 
+    @staticmethod
+    def parse_names(line: str) -> tuple[str, str]:
```

Comment:
Perhaps we could return a named tuple here? That might make it easier to see at-a-glance what the tuple _means_ semantically just from the function signature, without having to look into the details of the function implementation.

Something like this?

```diff
--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -4567,6 +4567,11 @@ class ParamState(enum.IntEnum):
     RIGHT_SQUARE_AFTER = 6


+class ParsedNames(NamedTuple):
+    full_name: str
+    c_basename: str
+
+
 class DSLParser:
     function: Function | None
     state: StateKeeper
@@ -4831,7 +4836,7 @@ def state_dsl_start(self, line: str) -> None:
         self.next(self.state_modulename_name, line)

     @staticmethod
-    def parse_names(line: str) -> tuple[str, str]:
+    def parse_names(line: str) -> ParsedNames:
         left, _as, right = line.partition(' as ')
         full_name = left.strip()
         c_basename = right.strip()
@@ -4846,7 +4851,7 @@ def parse_names(line: str) -> tuple[str, str]:
             fail(f"Illegal function name: {full_name!r}")
         if not is_legal_c_identifier(c_basename):
             fail(f"Illegal C basename: {c_basename!r}")
-        return full_name, c_basename
+        return ParsedNames(full_name=full_name, c_basename=c_basename)

     def update_function_kind(self, fullname: str) -> None:
```

We can maybe bikeshed the name of the namedtuple, though; `ParsedNames` probably isn't ideal...

Reply:
Yes, that's a good idea. Kinda like the `VersionTuple`.

Reply:
(Where's the "poll" option for GitHub posts 😆)

- [ ] ParsedNames
- [ ] NameTuple
- [ ] FunctionNameTuple
- [ ] Names
- [ ] FunctionNames

Reply:
`FunctionNames` is quite nice!

Reply:
Or maybe `ParsedFunctionNames`? I'm easy either way